# بسم الله الرحمن الرحيم

In [69]:
from utils import *
import pandas as pd 
from sklearn.model_selection import train_test_split


## Reading the Dataset, cleaning , preparing it for the training 

In [70]:
df = pd.read_csv("dataset/body_level_classification_train.csv")

In [71]:
df_original= cleaning_data(df).copy()

In [72]:
all_class_train,all_class_test=data_spliting(df_original,1/5)

In [73]:
oversampled= over_sampling(all_class_train,sampling_ratio=0.8)

In [74]:
X_train=oversampled.drop(["Body_Level"],axis=1).to_numpy()
Y_train=oversampled["Body_Level"].to_numpy()
X_test =all_class_test.drop(["Body_Level"],axis=1).to_numpy()
Y_test =all_class_test["Body_Level"].to_numpy()

In [75]:
X_train, scaler= data_scaling(X_train)
with open(f"scaler","wb") as fp : 
    pickle.dump(scaler,fp)

In [76]:
X_val, X_test, Y_val, Y_test = train_test_split(X_test
                                                     , Y_test, test_size=0.5, random_state=42)

In [77]:
X_val = scaler.transform(X_val)
X_test=scaler.transform(X_test)

In [78]:
model=build_model(2,0.01)

In [79]:
history = model.fit(X_train,Y_train,validation_data=[X_val,Y_val] 
             ,batch_size=32, epochs=400 , verbose =1, 
                       callbacks=[CustomCallback("NNModels/after_sampling/fifth_model.h5")])

Epoch 1/400
46/46 [==============================] - 1s 10ms/step - loss: 0.7707 - accuracy: 0.4844 - val_loss: 0.3377 - val_accuracy: 0.6735
Epoch 2/400
46/46 [==============================] - 0s 4ms/step - loss: 0.3616 - accuracy: 0.6857 - val_loss: 0.2382 - val_accuracy: 0.8027
Epoch 3/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2828 - accuracy: 0.7653 - val_loss: 0.2080 - val_accuracy: 0.8367
Epoch 4/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2437 - accuracy: 0.7837 - val_loss: 0.1859 - val_accuracy: 0.8299
Epoch 5/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2161 - accuracy: 0.8109 - val_loss: 0.1704 - val_accuracy: 0.8639
Epoch 6/400
46/46 [==============================] - 0s 2ms/step - loss: 0.1986 - accuracy: 0.8265 - val_loss: 0.1636 - val_accuracy: 0.8571
Epoch 7/400
46/46 [==============================] - 0s 2ms/step - loss: 0.1857 - accuracy: 0.8354 - val_loss: 0.1502 - val_accuracy: 0.8707
Epoch 8/400


46/46 [==============================] - 0s 3ms/step - loss: 0.0559 - accuracy: 0.9558 - val_loss: 0.0463 - val_accuracy: 0.9796
Epoch 42/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0541 - accuracy: 0.9605 - val_loss: 0.0460 - val_accuracy: 0.9796
Epoch 43/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0536 - accuracy: 0.9585 - val_loss: 0.0440 - val_accuracy: 0.9796
Epoch 44/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0530 - accuracy: 0.9612 - val_loss: 0.0426 - val_accuracy: 0.9864
Epoch 45/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0516 - accuracy: 0.9626 - val_loss: 0.0413 - val_accuracy: 0.9796
Epoch 46/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0506 - accuracy: 0.9619 - val_loss: 0.0435 - val_accuracy: 0.9796
Epoch 47/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0497 - accuracy: 0.9571 - val_loss: 0.0404 - val_accuracy: 0.9796
Epoch 48/400
46/46 

Epoch 64/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0377 - accuracy: 0.9701 - val_loss: 0.0288 - val_accuracy: 0.9864
Epoch 65/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0362 - accuracy: 0.9719
Yes You are here 0.9727891087532043 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:32        17192
config.json                                    2023-05-16 21:28:32         1588
metadata.json                                  2023-05-16 21:28:32           64
46/46 [===

46/46 [==============================] - 0s 3ms/step - loss: 0.0327 - accuracy: 0.9769 - val_loss: 0.0266 - val_accuracy: 0.9864
Epoch 76/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0319 - accuracy: 0.9782 - val_loss: 0.0276 - val_accuracy: 0.9796
Epoch 77/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0319 - accuracy: 0.9721 - val_loss: 0.0252 - val_accuracy: 0.9864
Epoch 78/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0313 - accuracy: 0.9755 - val_loss: 0.0250 - val_accuracy: 0.9864
Epoch 79/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0308 - accuracy: 0.9789 - val_loss: 0.0252 - val_accuracy: 0.9796
Epoch 80/400
25/46 [===============>..............] - ETA: 0s - loss: 0.0299 - accuracy: 0.9725
Yes You are here 0.9768707752227783 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean


46/46 [==============================] - 0s 3ms/step - loss: 0.0267 - accuracy: 0.9776 - val_loss: 0.0229 - val_accuracy: 0.9864
Epoch 93/400
37/46 [=======================>......] - ETA: 0s - loss: 0.0253 - accuracy: 0.9823
Yes You are here 0.9768707752227783 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:35        17192
config.json                                    2023-05-16 21:28:35         1588
metadata.json                                  2023-05-16 21:28:35           64
46/46 [================

Epoch 103/400
43/46 [===========================>..] - ETA: 0s - loss: 0.0236 - accuracy: 0.9797
Yes You are here 0.9795918464660645 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:36        17192
config.json                                    2023-05-16 21:28:36         1588
metadata.json                                  2023-05-16 21:28:36           64
46/46 [==============================] - 0s 2ms/step - loss: 0.0240 - accuracy: 0.9796 - val_loss: 0.0197 - val_accuracy: 0.9864
Epoch 104/400
46/46 [=

40/46 [=========================>....] - ETA: 0s - loss: 0.0228 - accuracy: 0.9789
Yes You are here 0.9789115786552429 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:38        17192
config.json                                    2023-05-16 21:28:38         1588
metadata.json                                  2023-05-16 21:28:38           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0237 - accuracy: 0.9789 - val_loss: 0.0195 - val_accuracy: 0.9864
Epoch 114/400
46/46 [===============

Epoch 124/400
41/46 [=========================>....] - ETA: 0s - loss: 0.0213 - accuracy: 0.9802
Yes You are here 0.9809523820877075 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:39        17192
config.json                                    2023-05-16 21:28:39         1588
metadata.json                                  2023-05-16 21:28:39           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0209 - accuracy: 0.9810 - val_loss: 0.0163 - val_accuracy: 0.9864
Epoch 125/400
28/46 [=

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:40        17192
config.json                                    2023-05-16 21:28:40         1588
metadata.json                                  2023-05-16 21:28:40           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0195 - accuracy: 0.9816 - val_loss: 0.0173 - val_accuracy: 0.9864
Epoch 134/400
41/46 [=========================>....] - ETA: 0s - loss: 0.0190 - accuracy: 0.9848
Yes You are here 0.9850339889526367 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

Epoch 159/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0176 - accuracy: 0.9823 - val_loss: 0.0146 - val_accuracy: 0.9864
Epoch 160/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0168 - accuracy: 0.9837 - val_loss: 0.0156 - val_accuracy: 0.9864
Epoch 161/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0173 - accuracy: 0.9871 - val_loss: 0.0143 - val_accuracy: 0.9864
Epoch 162/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0170 - accuracy: 0.9830 - val_loss: 0.0155 - val_accuracy: 0.9932
Epoch 163/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0167 - accuracy: 0.9830 - val_loss: 0.0133 - val_accuracy: 0.9864
Epoch 164/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0169 - accuracy: 0.9850 - val_loss: 0.0142 - val_accuracy: 0.9864
Epoch 165/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0159 - accuracy: 0.9857 - val_loss: 0.0137 - val_accuracy: 0.9864

variables.h5                                   2023-05-16 21:28:46        17192
config.json                                    2023-05-16 21:28:46         1588
metadata.json                                  2023-05-16 21:28:46           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0163 - accuracy: 0.9850 - val_loss: 0.0125 - val_accuracy: 0.9932
Epoch 184/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0135 - accuracy: 0.9865
Yes You are here 0.9850339889526367 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5          

Epoch 204/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0129 - accuracy: 0.9917
Yes You are here 0.9891156554222107 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:48        17192
config.json                                    2023-05-16 21:28:48         1588
metadata.json                                  2023-05-16 21:28:48           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0135 - accuracy: 0.9891 - val_loss: 0.0120 - val_accuracy: 0.9932
Epoch 205/400
26/46 [=

config.json                                    2023-05-16 21:28:50         1588
metadata.json                                  2023-05-16 21:28:50           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0129 - accuracy: 0.9891 - val_loss: 0.0115 - val_accuracy: 0.9932
Epoch 217/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0129 - accuracy: 0.9878 - val_loss: 0.0109 - val_accuracy: 0.9864
Epoch 218/400
36/46 [======================>.......] - ETA: 0s - loss: 0.0127 - accuracy: 0.9887
Yes You are here 0.9870748519897461 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                              

46/46 [==============================] - 0s 2ms/step - loss: 0.0128 - accuracy: 0.9871 - val_loss: 0.0122 - val_accuracy: 0.9864
Epoch 232/400
39/46 [========================>.....] - ETA: 0s - loss: 0.0127 - accuracy: 0.9864
Yes You are here 0.9863945841789246 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:53        17192
config.json                                    2023-05-16 21:28:52         1588
metadata.json                                  2023-05-16 21:28:52           64
46/46 [===============

.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:55        17192
config.json                                    2023-05-16 21:28:55         1588
metadata.json                                  2023-05-16 21:28:55           64
46/46 [==============================] - 0s 2ms/step - loss: 0.0124 - accuracy: 0.9878 - val_loss: 0.0108 - val_accuracy: 0.9932
Epoch 247/400
25/46 [===============>..............] - ETA: 0s - loss: 0.0114 - accuracy: 0.9900
Yes You are here 0.9884353876113892 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
........

31/46 [===================>..........] - ETA: 0s - loss: 0.0111 - accuracy: 0.9859
Yes You are here 0.9870748519897461 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:57        17192
config.json                                    2023-05-16 21:28:57         1588
metadata.json                                  2023-05-16 21:28:57           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0117 - accuracy: 0.9871 - val_loss: 0.0106 - val_accuracy: 0.9932
Epoch 259/400
46/46 [===============

Epoch 269/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0119 - accuracy: 0.9905 - val_loss: 0.0112 - val_accuracy: 0.9864
Epoch 270/400
39/46 [========================>.....] - ETA: 0s - loss: 0.0119 - accuracy: 0.9864
Yes You are here 0.9884353876113892 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:28:58        17192
config.json                                    2023-05-16 21:28:58         1588
metadata.json                                  2023-05-16 21:28:58           64
46/46 [=

31/46 [===================>..........] - ETA: 0s - loss: 0.0103 - accuracy: 0.9889
Yes You are here 0.9857142567634583 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:00        17192
config.json                                    2023-05-16 21:29:00         1588
metadata.json                                  2023-05-16 21:29:00           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0123 - accuracy: 0.9857 - val_loss: 0.0119 - val_accuracy: 0.9932
Epoch 280/400
29/46 [===============

Epoch 290/400
29/46 [=================>............] - ETA: 0s - loss: 0.0109 - accuracy: 0.9903
Yes You are here 0.9891156554222107 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:01        17192
config.json                                    2023-05-16 21:29:01         1588
metadata.json                                  2023-05-16 21:29:01           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0109 - accuracy: 0.9891 - val_loss: 0.0104 - val_accuracy: 0.9932
Epoch 291/400
30/46 [=

config.json                                    2023-05-16 21:29:03         1588
metadata.json                                  2023-05-16 21:29:03           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0102 - accuracy: 0.9912 - val_loss: 0.0108 - val_accuracy: 0.9932
Epoch 303/400
24/46 [==============>...............] - ETA: 0s - loss: 0.0090 - accuracy: 0.9922
Yes You are here 0.9897959232330322 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:03        17192
config.json           

Epoch 312/400
36/46 [======================>.......] - ETA: 0s - loss: 0.0101 - accuracy: 0.9922
Yes You are here 0.9897959232330322 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:05        17192
config.json                                    2023-05-16 21:29:05         1588
metadata.json                                  2023-05-16 21:29:05           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0106 - accuracy: 0.9898 - val_loss: 0.0088 - val_accuracy: 0.9932
Epoch 313/400
29/46 [=

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:06        17192
config.json                                    2023-05-16 21:29:06         1588
metadata.json                                  2023-05-16 21:29:06           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0101 - accuracy: 0.9857 - val_loss: 0.0118 - val_accuracy: 0.9932
Epoch 321/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0098 - accuracy: 0.9929
Yes You are here 0.9918367266654968 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

config.json                                    2023-05-16 21:29:07         1588
metadata.json                                  2023-05-16 21:29:07           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0100 - accuracy: 0.9878 - val_loss: 0.0118 - val_accuracy: 0.9932
Epoch 331/400
26/46 [===============>..............] - ETA: 0s - loss: 0.0102 - accuracy: 0.9880
Yes You are here 0.9911564588546753 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:08        17192
config.json           

46/46 [==============================] - 0s 4ms/step - loss: 0.0095 - accuracy: 0.9912 - val_loss: 0.0084 - val_accuracy: 0.9932
Epoch 341/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0098 - accuracy: 0.9912 - val_loss: 0.0096 - val_accuracy: 0.9864
Epoch 342/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0092 - accuracy: 0.9899
Yes You are here 0.9877551198005676 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:09        17192
config.json                            

config.json                                    2023-05-16 21:29:11         1588
metadata.json                                  2023-05-16 21:29:11           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0103 - accuracy: 0.9898 - val_loss: 0.0093 - val_accuracy: 0.9932
Epoch 353/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0093 - accuracy: 0.9925 - val_loss: 0.0108 - val_accuracy: 0.9864
Epoch 354/400
27/46 [================>.............] - ETA: 0s - loss: 0.0082 - accuracy: 0.9896
Yes You are here 0.9891156554222107 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                              

46/46 [==============================] - 0s 2ms/step - loss: 0.0092 - accuracy: 0.9898 - val_loss: 0.0092 - val_accuracy: 0.9932
Epoch 363/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0090 - accuracy: 0.9912 - val_loss: 0.0107 - val_accuracy: 0.9864
Epoch 364/400
25/46 [===============>..............] - ETA: 0s - loss: 0.0075 - accuracy: 0.9962
Yes You are here 0.9911564588546753 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:13        17192
config.json                            

Epoch 373/400
45/46 [============================>.] - ETA: 0s - loss: 0.0090 - accuracy: 0.9910
Yes You are here 0.9911564588546753 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:14        17192
config.json                                    2023-05-16 21:29:14         1588
metadata.json                                  2023-05-16 21:29:14           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0088 - accuracy: 0.9912 - val_loss: 0.0092 - val_accuracy: 0.9932
Epoch 374/400
30/46 [=

46/46 [==============================] - 0s 4ms/step - loss: 0.0096 - accuracy: 0.9905 - val_loss: 0.0108 - val_accuracy: 0.9932
Epoch 385/400
27/46 [================>.............] - ETA: 0s - loss: 0.0068 - accuracy: 0.9942
Yes You are here 0.9918367266654968 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:16        17192
config.json                                    2023-05-16 21:29:16         1588
metadata.json                                  2023-05-16 21:29:16           64
46/46 [===============

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:29:17        17192
config.json                                    2023-05-16 21:29:17         1588
metadata.json                                  2023-05-16 21:29:17           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0090 - accuracy: 0.9891 - val_loss: 0.0105 - val_accuracy: 0.9932
Epoch 395/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0085 - accuracy: 0.9899
Yes You are here 0.9904761910438538 0.9931972622871399
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

In [80]:
accurate_train_accuracy,classify_report= test_model(model,X_train,Y_train)

46/46 [==============================] - 0s 939us/step


In [81]:
accurate_train_accuracy

0.992517006802721

In [82]:
print(classify_report)

              precision    recall  f1-score   support

           0       0.98      1.00      0.99       272
           1       0.99      0.97      0.98       278
           2       0.99      0.99      0.99       376
           3       1.00      1.00      1.00       544

    accuracy                           0.99      1470
   macro avg       0.99      0.99      0.99      1470
weighted avg       0.99      0.99      0.99      1470



In [83]:
accurate_val_accuracy,classify_report= test_model(model,X_val,Y_val)

5/5 [==============================] - 0s 1ms/step


In [84]:
accurate_val_accuracy

0.9931972789115646

In [85]:
print(classify_report)

              precision    recall  f1-score   support

           0       1.00      0.95      0.97        20
           1       0.95      1.00      0.98        20
           2       1.00      1.00      1.00        36
           3       1.00      1.00      1.00        71

    accuracy                           0.99       147
   macro avg       0.99      0.99      0.99       147
weighted avg       0.99      0.99      0.99       147



In [86]:
accurate_test_accuracy,classify_report= test_model(model,X_test,Y_test)

5/5 [==============================] - 0s 1ms/step


In [87]:
accurate_test_accuracy

0.9932432432432432

In [67]:
print(classify_report)

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        18
           1       1.00      1.00      1.00        20
           2       1.00      0.98      0.99        45
           3       0.98      1.00      0.99        65

    accuracy                           0.99       148
   macro avg       1.00      0.99      1.00       148
weighted avg       0.99      0.99      0.99       148



## Notes

- name            : first model 
- split           : 1/3 
- oversampling    : 0.5 
- val_acc         : 0.9836734693877551
- test_acc        : 0.983739837398374
- gamma           : 1 
---------------------------------------------------------------------------------------------
- name            : second model 
- split           : 1/3 
- oversampling    : 0.8
- val_acc         : 0.9877551020408163
- test_acc        : 0.9878048780487805
- gamma           : 1 

---------------------------------------------------------------------------------------------

- name            : forth model 
- split           : 1/5
- oversampling    : 1
- val_acc         : 0.9863945578231292
- test_acc        : 0.9932432432432432
- gamma           : 2

---------------------------------------------------------------------------------------------
- name            : fifth model 
- split           : 1/5
- oversampling    : 0.5
- val_acc         : 0.9863945578231292
- test_acc        : 0.9864864864864865
- gamma           : 2

---------------------------------------------------------------------------------------------



In [68]:
df.columns


Index(['Gender', 'Age', 'Height', 'Weight', 'H_Cal_Consump', 'Veg_Consump',
       'Water_Consump', 'Alcohol_Consump', 'Smoking', 'Meal_Count',
       'Food_Between_Meals', 'Fam_Hist', 'H_Cal_Burn', 'Phys_Act',
       'Time_E_Dev', 'Transport', 'Body_Level'],
      dtype='object')